### DISCUSSION TITLE
How to set the M of HNSW in python sdk ?

### DISCUSSION THREAD

**Discussion by shiwanghua** [facebookresearch/faiss#3937] [Q&A]
https://github.com/facebookresearch/faiss/blob/af70c5bcce07eb419a7e6d33830f78b1e8413465/tests/test_graph_based.py

https://github.com/facebookresearch/faiss/blob/af70c5bcce07eb419a7e6d33830f78b1e8413465/tests/test_index.py

**Comment by shiwanghua** [comment]
refer to https://github.com/facebookresearch/faiss/blob/af70c5bcce07eb419a7e6d33830f78b1e8413465/faiss/impl/HNSW.cpp#L59

I set M by this way:
```
index = faiss.IndexHNSWFlat(d, 16, faiss.METRIC_INNER_PRODUCT)
index.hnsw.efConstruction = 256
index.hnsw.efSearch = 64
M=32
index.hnsw.set_default_probas(M, 1.0 / math.log(M));
index_with_ids = faiss.IndexIDMap2(index)
```

Is it right?

**Comment by ramilbakhshyiev** [comment]
Based on the code provided, I see that you specific M to 16 in the first line and then was to change it to 32. Can you please help me understand why you don't just set it to 32 from the start? Thanks!
